driver-core/src/main/java/com/datastax/driver/core/Connection.java

```diff
@@ -196,6 +196,9 @@ public void operationComplete(ChannelFuture future) throws Exception {
         Futures.addCallback(initFuture, new FutureCallback<Void>() {
             @Override
             public void onSuccess(Void result) {
+                Host host = factory.manager.metadata.getHost(address);
+                if(host != null)
```

Comment:
:+1: i'll look more into both the connection success/connection closed scenarios where the Host is gone and let you know what I find.